pkg/prometheus/promcfg.go

```diff
@@ -421,7 +423,7 @@ func compareScrapeTimeoutToScrapeInterval(scrapeTimeout, scrapeInterval monitori
 
 // Generate creates a serialized YAML representation of a Prometheus configuration using the provided resources.
 func (cg *ConfigGenerator) Generate(
```

Comment:
I know `ConfigGenerator` type already has a field called `spec` where we can extract `cpf`. Since we are refactoring a bit how hard is it to make this consistent? (I'm not sure, just raising the question) 
Either we could always the necessary structure (`PrometheusInterface`) as function arguments or have them as a type in `ConfigGenerator`

Reply:
Thanks for the initial review Joao! Could you elaborate a little bit here?

Do you mean changing the ConfigGenerator struct to something like this?
```go
type ConfigGenerator struct {
	logger                 log.Logger
	version                semver.Version
	notCompatible          bool
	//spec                   *v1.PrometheusSpec
	spec                   *v1.PrometheusInterface
	endpointSliceSupported bool
}
```

Reply:
I think that it makes sense to have the PrometheusInterface field in the config generator. Any parameter that is specific to the Prometheus server can be passed to the config generator's methods.
E.g. [appendStorageSettingsConfig](https://github.com/prometheus-operator/prometheus-operator/blob/fcd34a44045503bc45093887776c4b9ca675f6ed/pkg/prometheus/promcfg.go#L578-L607) would become

```golang
func (cg *ConfigGenerator) appendStorageSettingsConfig(cfg yaml.MapSlice, exemplars *v1.Exemplars, tsdb v1.TSBD) (yaml.MapSlice, error) {
...
}
```